### PR TITLE
Move status.proto to server/status

### PR DESCRIPTION
**Pull request by tamird**
Lots of juggling to break circular dependencies.

### CODE REVIEW

**Comment by tamird** [comment]
@petermattis gonna rename MVCCStats as you suggested, it really does not belong in /engine

**Comment by MBerhault** [comment]
LGTM

**Comment by spencerkimball** [comment]
I hope these hoops you're jumping through aren't going to be the new normal as we split up the protos. I'm not super comfortable with storage and storage/engine importing server/status. Sigh.

File: server/status/feed.go [hunk 1]
```diff
@@ -22,6 +22,79 @@ import (
 	"github.com/cockroachdb/cockroach/util"
 )
 
+// RegisterRangeEvent occurs in two scenarios. Firstly, while a store
+// broadcasts its list of ranges to initialize one or more new accumulators
+// (with Scan set to true), or secondly, when a new range is initialized on
+// the store (for example through replication), with Scan set to false. This
+// event includes the Range's RangeDescriptor and current StoreStats.
+type RegisterRangeEvent struct {
```
Comment (bdarnell) [inline server/status/feed.go]: I don't like the move of feed events from storage to server/status. This seems to point to a future in which server/status contains events from many feed publishers and becomes a sprawling mess like the monolithic proto package. These events (and the StoreStats proto) seem to belong closer to storage (perhaps a new storage/storestatus package if the relationship between storage and storage/engine makes things tricky).

Comment (tamird) [inline server/status/feed.go]: This is a good point. I'll see what I can do.

**Comment by tamird** [comment]
Redid this, PTAL

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**